File: tests/output/focus/src/model/restaurant/paiement-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_DATE_HEURE, DO_ID, DO_PRIX} from "../../domains";

export type PaiementItem = EntityToType<PaiementItemEntityType>;
export type PaiementItemEntityType = typeof PaiementItemEntity;

export const PaiementItemEntity = entity({
    factureId: e.field(DO_ID, f => f
        .label("restaurant.paiement.factureId")
    ),
    swileCardId: e.field(DO_ID, f => f
        .label("restaurant.paiement.swileCardId")
    ),
    dateCommande: e.field(DO_DATE_HEURE, f => f
        .label("restaurant.commande.dateCommande")
    ),
    montantTotal: e.field(DO_PRIX, f => f
        .label("restaurant.commande.montantTotal")
    )
});

File: tests/output/jpa/src/main/javagen/restaurant/jpa_server/dtos/restaurant/PaiementItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_server.dtos.restaurant;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.annotation.Generated;
import jakarta.validation.constraints.NotNull;

/**
 * Détail d'un paiement.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public record PaiementItem(
	@NotNull Integer factureId,
	@NotNull Integer swileCardId,
	@NotNull LocalDateTime dateCommande,
	@NotNull BigDecimal montantTotal
) {
}
